### PR TITLE
actix-rt: expose JoinError

### PR DESCRIPTION
**Pull request by asonix**
## PR Type
Feature? Is it a bug if nothing's technically broken?

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview
Add `JoinError` to the public exports in the task module

Closes https://github.com/actix/actix-net/issues/270

### CODE REVIEW

**Comment by asonix** [comment]
CI failures look like https://github.com/hyperium/hyper/issues/2421

File: actix-rt/CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changes
 
 ## Unreleased - 2021-xx-xx
+* Expose JoinError from tokio
```
Comment (robjtede) [inline actix-rt/CHANGES.md:4]:
```suggestion
* Expose JoinError from tokio. [#271]

[#271]: https://github.com/actix/actix-net/pull/271
```

**Review by robjtede — APPROVED** [review]

**Comment by robjtede** [comment]
Thanks @asonix 🚀

**Comment by robjtede** [comment]
released as 2.0.1

**Pull request merged**